pkg/addonmanager/addonclients/upgrader.go

```diff
@@ -55,7 +55,7 @@ func (f *FluxAddonClient) Upgrade(ctx context.Context, managementCluster *types.
 	return types.NewChangeDiff(changeDiff), nil
 }
 
-func (f *FluxAddonClient) fluxChangeDiff(currentSpec, newSpec *cluster.Spec) *types.ComponentChangeDiff {
+func FluxChangeDiff(currentSpec, newSpec *cluster.Spec) *types.ComponentChangeDiff {
```

Comment:
I would maybe suggest returning `types.ChangeDiff` from the public method for all core components
That gives us more freedom to add/remove subcomponents from `Flux`, `EKS-A`, etc.
Also that makes the method signature consistent with the `Upgrade` method

Reply:
done.